### PR TITLE
Feature/persistent queries

### PR DESCRIPTION
**Pull request by godanny86**
## Description

Includes additional persistent query examples for Content Fragments:

* `adventure-by-path` - Returns an adventure based on a path
* `adventure-by-price` - Filters an adventure list based on a price and a given operator (greater than, less than, equal...)
* `adventure-by-slug` - Filters an adventure list based on a slug (shortened unique identifier)
* `adventures-all` - Returns all adventures
* `article-by-author` - Filters an article list based on the author's first and last name (separate parameters)
* `article-by-path` - Returns an article based on a path to a fragment
* `articles-all` - Returns all articles

## Motivation and Context

AEM Headless best practices should use Persistent queries whenever possible. This shows several examples of using persistent queries against WKND content.

## Screenshots (if appropriate):

image

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: ui.content.sample/src/main/content/jcr_root/conf/wknd/settings/graphql/persistentQueries/article-by-slug/_jcr_content/_jcr_data.binary [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+query getArticleBySlug($slug: String!) {
+  articleList(filter: {
+    slug: {
+      _expressions: [ { value: $slug } ]
+    }
+  	}) {
+    items {
+      _path
+      title
+      main {
```
Comment (davidjgonzalez) [inline ui.content.sample/src/main/content/jcr_root/conf/wknd/settings/graphql/persistentQueries/article-by-slug/_jcr_content/_jcr_data.binary:19]:
```suggestion
      title
      slug
      main {
```

File: ui.content.sample/src/main/content/jcr_root/conf/wknd/settings/graphql/persistentQueries/article-by-path/_jcr_content/_jcr_data.binary [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+# Expected query variables like:
+
+# includes a reference to the Author Model (nested fragment)
+
+query getArticleByPath($articlePath: String!){
+  articleByPath(_path: $articlePath) {
+    item {
+      _path
+      title
+      main {
```
Comment (davidjgonzalez) [inline ui.content.sample/src/main/content/jcr_root/conf/wknd/settings/graphql/persistentQueries/article-by-path/_jcr_content/_jcr_data.binary:11]:
```suggestion
      title
      slug
      main {
```

File: ui.content.sample/src/main/content/jcr_root/conf/wknd/settings/graphql/persistentQueries/article-by-author/_jcr_content/_jcr_data.binary [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+        }
+        lastName: {
+          _expressions: [ { value: $authorLastName } ]
+        }
+      }
+  	}) {
+    items {
+      _path
+      title
+      main {
```
Comment (davidjgonzalez) [inline ui.content.sample/src/main/content/jcr_root/conf/wknd/settings/graphql/persistentQueries/article-by-author/_jcr_content/_jcr_data.binary:20]:
```suggestion
      _path
      title
      slug
      main {
```

**Review by davidjgonzalez — APPROVED** [review]
Added `slug` to article queries

File: ui.content/src/main/content/jcr_root/conf/wknd/settings/graphql/persistentQueries/article-by-author/_jcr_content/_jcr_data.binary [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+          _expressions: [ { value: $authorFirstName } ]
+        }
+        lastName: {
+          _expressions: [ { value: $authorLastName } ]
+        }
+      }
+  	}) {
+    items {
+      _path
+      title
```
Comment (godanny86) [inline ui.content/src/main/content/jcr_root/conf/wknd/settings/graphql/persistentQueries/article-by-author/_jcr_content/_jcr_data.binary:19]:
```suggestion
      _path
      title
      slug
```

File: ui.content/src/main/content/jcr_root/conf/wknd/settings/graphql/persistentQueries/article-by-path/_jcr_content/_jcr_data.binary [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+# Expected query variables like:
+
+# includes a reference to the Author Model (nested fragment)
+
+query getArticleByPath($articlePath: String!){
+  articleByPath(_path: $articlePath) {
+    item {
+      _path
+      title
+      main {
```
Comment (godanny86) [inline ui.content/src/main/content/jcr_root/conf/wknd/settings/graphql/persistentQueries/article-by-path/_jcr_content/_jcr_data.binary:11]:
```suggestion
      _path
      title
      slug
      main {
```

**Pull request merged**